src/nvim/getchar.c

```diff
@@ -1065,7 +1065,7 @@ void del_typebuf(int len, int offset)
     /* adjust typebuf.tb_buf (include the NUL at the end) */
     memmove(typebuf.tb_buf + typebuf.tb_off + offset,
         typebuf.tb_buf + i + len,
-        (size_t)(typebuf.tb_len - offset + 1));
+        (size_t)typebuf.tb_len - (size_t )offset + 1);
```

Comment:
what if `offset` is negative? for these cases it's best to use `STRICT_ADD` (see existing usages for examples).